File: app/src/main/java/org/antrack/app/libs/LoadingDialog.java
package org.antrack.app.libs;

import android.app.Activity;
import android.app.ProgressDialog;

public class LoadingDialog {
    private static ProgressDialog loadingDialog;

    public static void show(final Activity act, final String message) {
        act.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                loadingDialog = new ProgressDialog(act);
                loadingDialog.setMessage(message);
                loadingDialog.setCancelable(false);
                loadingDialog.setInverseBackgroundForced(false);
                loadingDialog.show();
            }
        });
    }

    public static void hide(final Activity act) {
        act.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                if (loadingDialog != null) {
                    loadingDialog.hide();
                    // GC
                    loadingDialog = null;
                }
            }
        });
    }

    public static boolean isShown() {
        return loadingDialog != null;
    }

}

File: app/src/main/java/org/antrack/app/libs/LoadingDialog.kt
package org.antrack.app.libs

import android.app.Activity
import android.app.ProgressDialog

object LoadingDialog {
    lateinit var loadingDialog: ProgressDialog

    var isShown: Boolean = false

    fun show(act: Activity, message: String) {
        act.runOnUiThread {
            loadingDialog = ProgressDialog(act)
            loadingDialog.setMessage(message)
            loadingDialog.setCancelable(false)
            loadingDialog.setInverseBackgroundForced(false)
            loadingDialog.show()
            isShown = true
        }
    }

    fun hide(act: Activity) {
        act.runOnUiThread {
            if (isShown) {
                loadingDialog.hide()
                isShown = false
            }
        }
    }
}
